rswag-specs/lib/rswag/specs/request_factory.rb

```diff
@@ -173,6 +173,8 @@ def build_query_string_part(param, value, swagger_doc)
                           end
               return "#{escaped_name}=" + value.to_a.flatten.map{|v| CGI.escape(v.to_s) }.join(separator)
             end
+          when :string
+            return "#{escaped_name}=#{CGI.escape(value.to_s)}"
           else
             return "#{name}=#{value}"
```

Comment:
Why not just change this line instead? That way you'll never be adding unescaped values.